### PR TITLE
Northstar php  enablealizationing

### PR DESCRIPTION
**Pull request by weerd**
#### What's this PR do?

This PR incorporates Northstar-PHP into Phoenix. It sets up the package, including the repository for the client credentials tokens. It does not yet begin the process of switching out the old code in **dosomething_northstar** module from using `_dosomething_northstar_build_http_client()` to build the client and `drupal_http_request()` to make the HTTP calls. This will happen in an upcoming PR to help keep the focus of this one to just enabling the use of Northstar-PHP from within Phoenix.
#### How should this be reviewed?

👀 
#### Relevant tickets

Fixes https://github.com/DoSomething/TeamRocket/issues/12
#### Checklist
- [ ] Tested on staging.

---

@DFurnes @angaither 

@jessleenyc @ngjo

### CODE REVIEW

File: lib/app/Auth/PhoenixOAuthRepository.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    return NULL;
+  }
+
+  /**
+   * Get the OAuth client's token.
+   *
+   * @return \League\OAuth2\Client\Token\AccessToken|null
+   */
+  public function getClientToken() {
+    $clientId = variable_get('dosomething_northstar_oauth_client_token_clientid', NULL);
```
Comment (DFurnes) [inline lib/app/Auth/PhoenixOAuthRepository.php]: This variable looks unused.

File: lib/app/Auth/PhoenixOAuthRepository.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   * Save the access token for an authorized client.
+   *
+   * @param  string $clientId
+   * @param  string $accessToken
+   * @param  integer $expiration
+   * @param  string $role
+   * @return void
+   */
+  public function persistClientToken($clientId, $accessToken, $expiration, $role) {
+    variable_set('dosomething_northstar_oauth_client_token_clientid', $clientId);
```
Comment (DFurnes) [inline lib/app/Auth/PhoenixOAuthRepository.php]: See above.

**Review by DFurnes — CHANGES_REQUESTED** [review]
Two Qs, otherwise looks good!

File: lib/app/Auth/PhoenixOAuthRepository.php (hunk 1 above)
Comment (weerd) [reply lib/app/Auth/PhoenixOAuthRepository.php]: Yeah, I was referencing the example from the `LaravelOAuthRepository` in the NS-PHP package, and kinda realized that's only really used as an id to find the record in the DB... wasn't sure if we'd _want_/_need_ it here in Phoenix land.

**Review by DFurnes — APPROVED** [review]
🚢 🌠 🚀

**Pull request merged**